### PR TITLE
Code coverage increase

### PR DESCRIPTION
**Pull request by paplorinc**
Following the work of @AlparSzabados and @ruslansennov, here's my attempt to increase the code coverage :)

Improves: #1279

### CODE REVIEW

File: javaslang/src/test/java/javaslang/control/OptionTest.java [hunk 1]
```diff
@@ -156,25 +167,25 @@ public void shouldThrowOnGetWhenValueIsNotDefined() {
 
     @Test
     public void shouldReturnSelfOnOrElseIfValueIsPresent() {
-        Option<Integer> opt = Option.of(42);
+        final Option<Integer> opt = Option.of(42);
```
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/control/OptionTest.java]: +1 for `42`:P

File: javaslang/src/test/java/javaslang/control/OptionTest.java [hunk 2]
```diff
@@ -267,15 +278,10 @@ public void shouldThrowNullPointerExceptionWhenNullOnEmptyActionPassed() {
 
     @Test
     public void shouldThrowExceptionIfOnEmptySetAndOptionIsEmpty() {
```
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/control/OptionTest.java]: It's not throwing so it should be renamed.

File: javaslang/src/test/java/javaslang/control/OptionTest.java [hunk 3]
```diff
@@ -449,15 +455,15 @@ public void shouldNotConsumeAnythingOnPeekWhenValueIsNotDefined() {
 
     @Test(expected = NullPointerException.class)
     public void shouldThrowExceptionOnNullTransformFunction() {
-        Option<Integer> option = Option.some(1);
+        final Option<Integer> option = Option.some(1);
```
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/control/OptionTest.java]: you could inline this

File: javaslang/src/test/java/javaslang/collection/BitSetTest.java [hunk 4]
```diff
@@ -500,6 +497,135 @@ private void assertMinimumsAreEqual(java.util.BitSet oldSet, BitSet<Integer> new
         }
     }
 
+    // -- zip
```
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/collection/BitSetTest.java]: Great tests 👍

**Comment by paplorinc** [comment]
@codecov-io is lazy today: `0.24%` increase, i.e. we're at `97.73%`
https://codecov.io/gh/javaslang/javaslang/commit/ffe6583c3df393141fab993c1d924a7c6ad190ed

**Comment by AlparSzabados** [comment]
@paplorinc .23% to go. I will look for other potential victims 😈

File: javaslang/src/main/java/javaslang/control/Option.java [hunk 5]
```diff
@@ -159,7 +159,7 @@
     @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
     static <T> Option<T> ofOptional(Optional<? extends T> optional) {
         Objects.requireNonNull(optional, "optional is null");
-        return optional.isPresent() ? of(optional.get()) : none();
+        return optional.<Option<T>>map(Option::of).orElseGet(Option::none);
     }
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/control/Option.java]: 👍

File: javaslang/src/main/java/javaslang/collection/Iterator.java [hunk 6]
```diff
@@ -2068,9 +2068,6 @@ public T getNext() {
         public Object next() { throw new NoSuchElementException(stringPrefix() + ".next()"); }
 
         @Override
-        public void remove() { throw new IllegalStateException(stringPrefix() + ".remove()"); }
```
Comment (danieldietrich) [inline javaslang/src/main/java/javaslang/collection/Iterator.java]: yes, we've misinterpreted the ju.Iterator API before

**Pull request merged**